src/main/java/ui/issuepanel/PanelControl.java

```diff
@@ -191,15 +192,20 @@ public void closeCurrentPanel() {
         if (currentlySelectedPanel.isPresent()) {
             int panelIndex = currentlySelectedPanel.get();
             closePanel(panelIndex);
-            if (getChildren().size() == 0) {
-                setCurrentlySelectedPanel(Optional.empty());
-            } else {
-                int newPanelIndex = (panelIndex > getChildren().size() - 1)
-                                     ? panelIndex - 1
-                                     : panelIndex;
-                setCurrentlySelectedPanel(Optional.of(newPanelIndex));
-                getPanel(currentlySelectedPanel.get()).requestFocus();
-            }
+        }
+    }
+    
+    public void setFocusToPreviousPanel(int closedPanelIndex) {
+        if (getChildren().size() == 0) {
+            setCurrentlySelectedPanel(Optional.empty());
+        } else if (closedPanelIndex != currentlySelectedPanel.get()) {
+            return;
```

Comment:
This check should come first, as a guard clause. That way it's clear that this method does nothing if it wasn't the currently-selected panel that was closed.